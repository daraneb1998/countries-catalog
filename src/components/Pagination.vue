<script setup lang="ts">
import { useSearchStore } from "@/stores/search";
import { SortType } from "@/types";

const searchStore = useSearchStore();
const onGoToNextPage = () => {
  searchStore.onGoToNextPage();
};
const onGoToPreviousPage = () => {
  searchStore.onGoToPreviousPage();
};
</script>

<template>
  <div class="flex gap-x-2">
    <p class="text-gray-500 self-center text-base hidden md:inline">
      Page {{ searchStore.currentPageNumber }} ({{ searchStore.pageSize }}) of
      {{ searchStore.totalPages }}
    </p>
    <button
      :disabled="
        searchStore.currentPageNumber === 1 || searchStore.searchKeyword !== ''
      "
      class="border border-blue-500 hover:text-white px-4 hover:bg-blue-500 transition-colors disabled:bg-slate-200 disabled:text-gray-500 disabled:cursor-not-allowed"
      :class="
        searchStore.sortType === SortType.ASCENDING
          ? 'bg-blue-500 text-white'
          : 'bg-transparent'
      "
      @click="onGoToPreviousPage"
    >
      PREVIOUS
    </button>
    <button
      :disabled="
        searchStore.totalPages === searchStore.currentPageNumber ||
        searchStore.searchKeyword !== ''
      "
      class="border border-blue-500 hover:text-white px-4 hover:bg-blue-500 transition-colors disabled:bg-slate-200 disabled:text-gray-500 disabled:cursor-not-allowed"
      :class="
        searchStore.sortType === SortType.DESCENDING
          ? 'bg-blue-500 text-white'
          : 'bg-transparent'
      "
      @click="onGoToNextPage"
    >
      NEXT
    </button>
  </div>
</template>
