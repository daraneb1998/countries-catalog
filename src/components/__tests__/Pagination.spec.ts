import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import Pagination from "../Pagination.vue";

describe("Pagination", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it("renders properly", () => {
    const wrapper = mount(Pagination);
    expect(wrapper.text()).toMatchSnapshot();
  });
});
