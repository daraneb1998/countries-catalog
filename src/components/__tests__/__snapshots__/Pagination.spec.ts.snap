// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > renders properly 1`] = `""`;

exports[`Pagination > renders properly 1`] = `"Page 1 (25) of 0 PREVIOUS  NEXT"`;
